public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -91,6 +91,10 @@ export class AppChromeService {
 
   public setReturnToPrevious = (returnToPrevious: ReturnToPreviousProps) => {
     const previousPage = this.state.getValue().returnToPrevious;
+    const isReturnToPreviousEnabled = config.featureToggles.returnToPrevious;
+    if (!isReturnToPreviousEnabled) {
+      return;
+    }
```

Comment:
I think we should:

 - move this up one line (no need to get `previousPage` only to bail out if the toggle isnt set)
 - also add the same check to `clearReturnToPrevious` for thoroughness